ARROW-9358: [Integration] remove generated_large_batch.json

lidavidm opened the pull request:
This should speed up integration tests by moving the expensive large batch test to the individual Flight implementations.

pitrou commented on cpp/src/arrow/flight/flight_test.cc:
```diff
@@ -1105,6 +1110,15 @@ TEST_F(TestFlightClient, DoGetDicts) {
+  BatchVector expected_batches;
+  ASSERT_OK(ExampleLargeBatches(&expected_batches));
+  Ticket ticket{"ticket-large-batch-1"};
+  CheckTicket(ticket, expected_batches);
```
I was a bit confused here because I didn't understand that `CheckTicket` actually issued a GET request. Can you call it `CheckDoGet` instead?

pitrou commented on cpp/src/arrow/flight/flight_test.cc:
```diff
@@ -1515,6 +1529,26 @@ TEST_F(TestDoPut, DoPutDicts) {
+  const auto array_length = 32768;
+  auto descr = FlightDescriptor::Path({"floats"});
+  BatchVector batches;
+  std::vector<std::shared_ptr<arrow::Array>> arrays;
```
Can you delegate the record batch creation to a helper function, since it's done both for DoGet and DoPut tests?

pitrou commented:
cc @emkornfield for the Java side.

emkornfield commented:
Is there an urgency for this?  Have my hands full at work today and tomorrow at least, i can try to look later in the week?

pitrou commented:
No urgency at all. This is just an improvement in development comfort.

emkornfield commented:
@lidavidm this looks fine to me, but based on PR description I would have expected java to have some sort of file read removed?  Maybe its late but i just see added generation data which looks fine.

lidavidm commented:
Addressed feedback and rebased.

@emkornfield this PR _removes_ generation of a large data file, which would have been read by Java later, and then adds a Java test that generates the data in-memory instead to replace it.

emkornfield commented:
Thanks.  +1 merging.
